docs/src/tutorial/cython_tutorial.rst

```diff
@@ -132,27 +132,83 @@ them as a Python list.
     :linenos:
 
 You'll see that it starts out just like a normal Python function definition,
-except that the parameter ``kmax`` is declared to be of type ``int`` . This
+except that the parameter ``nb_primes`` is declared to be of type ``int`` . This
 means that the object passed will be converted to a C integer (or a
 ``TypeError.`` will be raised if it can't be).
 
+Now, let's dig into the core of the function::
+
+    cdef int n, i, len_p
+    cdef int p[1000]
+
 Lines 2 and 3 use the ``cdef`` statement to define some local C variables.
-Line 4 creates a Python list which will be used to return the result. You'll
-notice that this is done exactly the same way it would be in Python. Because
-the variable result hasn't been given a type, it is assumed to hold a Python
-object.
+The result is put in ``p``, it will be converted to a python list at the end
+of the function (line 22). ::
+
+    if nb_primes > 1000:
+        nb_primes = 1000
+
+As in C, declaring a static array requires knowing the size at compile time.
+We make sure the user doesn't set a value above 1000 (or we'll have a nice
```

Comment:
"we'll" sounds like you get a segfault also in this case, so "we would" seems better. And calling it a "nice" segfault seems a bit ... early at this stage, I'd say. Remember, not everyone who reads this tutorial knows C.